### PR TITLE
Update spring-boot version to 3.1.1

### PR DESCRIPTION
**Pull request by DanielFran**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/generator-constants.mts [hunk 1]
```diff
@@ -25,20 +25,20 @@ export const JAVA_COMPATIBLE_VERSIONS = ['17', '18', '19'];
...
    */
-  'spring-boot': '3.1.0',
+  'spring-boot': '3.1.1',
   /*
    * hibernate version should match the one managed by https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-dependencies/SPRING_BOOT_VERSION
    * Required due to hibernate-jpamodelgen annotation processor.
    */
   // TODO update to spring-boot provided version. Refer to https://github.com/jhipster/generator-jhipster/issues/22579
-  hibernate: '6.1.7.Final',
+  hibernate: '6.1.7.Final', // Should be 6.2.5.Final
```
Comment (mraible) [inline generators/generator-constants.mts:41]: Is there a bug in Hibernate 6.2 to track this or is it something we need to change?

Comment (DanielFran) [reply generators/generator-constants.mts:41]: Related with https://github.com/jhipster/generator-jhipster/issues/22579

File: generators/init/constants.mjs [hunk 2]
```diff
@@ -16,7 +16,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-export const NODE_VERSION = '14.18.1';
+export const NODE_VERSION = '18.16.1';
```
Comment (DanielFran) [inline generators/init/constants.mjs:19]: @mshima We should use only 1 constant for NODE_VERSION, no?
Maybe use only the one from init generator?

Comment (mshima) [reply generators/init/constants.mjs:19]: NODE_VERSION at init is used as required node version:
https://github.com/jhipster/generator-jhipster/blob/61ec0be217e40fa96c944bdc624921efb5104bb6/generators/init/templates/package.json.ejs#L35

I think it should match https://github.com/jhipster/generator-jhipster/blob/61ec0be217e40fa96c944bdc624921efb5104bb6/package.json#L179.
We probably should copy the package.json version.

File: generators/generator-constants.mts (hunk 1 above)
Comment (BrayanMnz) [reply generators/generator-constants.mts:41]: May I ask if updating to 6.2.5 still failing?

**Pull request merged**